gate-web/src/main/groovy/com/netflix/spinnaker/gate/security/x509/X509AuthenticationUserDetailsService.groovy

```diff
@@ -50,8 +53,11 @@ class X509AuthenticationUserDetailsService implements AuthenticationUserDetailsS
     def x509 = (X509Certificate) token.credentials
     def email = emailFromSubjectAlternativeName(x509) ?: token.principal
 
-    return new User(email: email,
-                    allowedAccounts: anonymousAccountsService.allowedAccounts)
+    return new User(
+      email: email,
+      allowedAccounts: anonymousAccountsService.allowedAccounts,
+      roles: userRolesProvider.loadRoles((String) email)
```

Comment:
You should use coercion instead of straight casting: `email as String`.

http://groovy-lang.org/operators.html#_coercion_operator